maitake-sync/src/blocking.rs

```diff
@@ -0,0 +1,165 @@
+//! Synchronous (blocking) synchronization primitives.
+//!
+//! The core synchronization primitives in `maitake-sync`, such as
+//! [`Mutex`](crate::Mutex),  [`RwLock`](crate::RwLock), and
+//! [`WaitQueue`](crate::WaitQueue) are _asynchronous_. They are designed to be
+//! used with [`core::task`] and [`core::future`], and when it is necessary to
+//! wait for another task to complete some work for the current task to proceed,
+//! `maitake`'s synchronization primitives wait by *yielding* to the
+//! asynchronous task scheduler to allow another task to proceed.
+//!
+//! This module, on the other hand, provides _synchronous_ (or _blocking_)
+//! synchronization primitives. Rather than yielding to the runtime, these
```

Comment:
synchronousization

Reply:
wait, i think it's spelled correctly here?

Reply:
oh you were just doing a joke...